src/pip/_internal/utils/misc.py

```diff
@@ -907,6 +911,18 @@ def remove_auth_from_url(url):
     return surl
 
 
+def remove_auth_from_url(url):
+    # Return a copy of url with 'username:password@' removed.
+    # username/pass params are passed to subversion through flags
+    # and are not recognized in the url.
+    return transform_url(url, lambda netloc: split_auth_from_netloc(netloc)[0])
```

Comment:
I would define the lambda on its own line:
`transform_netloc = lambda netloc: split_auth_from_netloc(netloc)[0]`

Reply:
The linter dislikes this, and I also try not to do this. I think it's fine as it is?

Reply:
What does the linter not like? I’m not a fan of cramming a lot into one line. You can also define the function using def syntax.

Reply:
Lambdas are supposed to be anonymous - if you bind them to a name it kind of defeats the point. The linter points this out and tells you to use `def`.

I'll do whatever you suggest (I don't care that much), but a lambda seems perfect here. A whole named function for this seems overkill.

Reply:
I just think the code will be easier to understand if you give the function a name (whether that be a lambda or def syntax). It will also have a more parallel / symmetric look with `redact_password_from_url` below.